core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -29,9 +30,9 @@
  *
  * @param <T> the type of resource this pool manages
  */
-// TODO(peis): Implements this from the Pool interface.
 @ThreadSafe
-public abstract class ResourcePool<T> {
+public abstract class ResourcePool<T> implements Pool<T> {
```

Comment:
@peisun1115 I think this abstract class should be renamed to `ConcurrentResourcePool`, and the original interface `Pool` might be more accurately called `ResourcePool` if you check out its description.